{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# find\n",
    "\n",
    "Syntax to find all .txt files in a directory,\n",
    "\n",
    "```\n",
    "find ./<dir-name> -name \"*.txt\"\n",
    "```\n",
    "\n",
    "Syntax to execute a command on all the files that match a pattern.\n",
    "\n",
    "```\n",
    "find ./<dir-name> -name \"*.txt\" -exec bash -c 'wc -l $1' _ {} \\;\n",
    "```\n",
    "\n",
    "# Git\n",
    "\n",
    "Git Commands,\n",
    "\n",
    "```\n",
    "cd test-repo/\n",
    "git init # Start a git repository\n",
    "git add . # Adds files into the git index\n",
    "git commit -m \"Message here\" # Takes a snapshot of the index\n",
    "git push <remote-url> master # Pushed commits to a remote repository\n",
    "git pull # Pulls from a remote repository\n",
    "```\n",
    "\n",
    "# String Replacement\n",
    "\n",
    "Command to replace pattern in string,\n",
    "\n",
    "```\n",
    "a=\"helloworld\"\n",
    "a=${a/world/hello}\n",
    "echo $a\n",
    "```\n",
    "\n",
    "# FASTQs\n",
    "\n",
    "Command to extract all read names from FASTQ files,\n",
    "\n",
    "```\n",
    "awk 'NR%4==1' data/tmp/simulated1.R1.fq\n",
    "```\n",
    "\n",
    "## Questions\n",
    "\n",
    "1. Count number of lines in each .txt files in ./data/tmp/ folder.\n",
    "2. Create a local `git` repository on your account. Add atleast two files and make a commit. \n",
    "3. Push these changes to a remote repository and add the url here.\n",
    "4. Print all nucleotide sequences from data/tmp/simulate1.R1.fq\n",
    "\n",
    "## HW\n",
    "\n",
    "1. Count number of characters in the filename of each .txt file in ./data/tmp/ folder.\n",
    "3. Print all nucleotide sequences from ./data/fastq/simulated1.R1.fq by replacing all 'T's in the nucleotide sequences with 'U's. Hint: ${variable//123/112} will replace all occurences of 123 with 112."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "findÂ¶\n",
    "\n",
    "Syntax to find all .txt files in a directory,\n",
    "\n",
    "find ./<dir-name> -name \"*.txt\"\n",
    "\n",
    "Syntax to execute a command on all the files that match a pattern.\n",
    "\n",
    "find ./<dir-name> -name \"*.txt\" -exec bash -c 'wc -l $1' _ {} \\"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Git\n",
    "Git Commands,\n",
    "\n",
    "cd test-repo/\n",
    "git init # Start a git repository\n",
    "git add . # Adds files into the git index\n",
    "git commit -m \"Message here\" # Takes a snapshot of the index\n",
    "git push <remote-url> master # Pushed commits to a remote repository\n",
    "git pull # Pulls from a remote repository"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1.3_jupyter-basics.ipynb\n",
      "1.4_working-with-files.ipynb\n",
      "1.5_redirection-and-pipes.ipynb\n",
      "1.6_awk.ipynb\n",
      "1.7_for_loop_string_replacement.ipynb\n",
      "1.8_find_git_stringreplacement_questions.ipynb\n",
      "Applied-Bioinformatics\n",
      "data\n",
      "file_list.txt\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "bash: line 1: cd: test-repo: No such file or directory\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd test-repo\n",
    "ls"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "String Replacement\n",
    "Command to replace pattern in string,\n",
    "\n",
    "a=\"helloworld\"\n",
    "a=${a/world/hello}\n",
    "echo $a"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "FASTQs\n",
    "Command to extract all read names from FASTQ files,\n",
    "\n",
    "awk 'NR%4==1' data/tmp/simulated1.R1.fq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Questions\n",
    "Count number of lines in each .txt files in ./data/tmp/ folder.\n",
    "Create a local git repository on your account. Add atleast two files and make a commit.\n",
    "Push these changes to a remote repository and add the url here.\n",
    "Print all nucleotide sequences from data/tmp/simulate1.R1.fq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reinitialized existing Git repository in /home/bbfxb2/Applied-Bioinformatics-HW/.git/\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "error: unknown switch `f'\n",
      "usage: git commit [<options>] [--] <pathspec>...\n",
      "\n",
      "    -q, --quiet           suppress summary after successful commit\n",
      "    -v, --verbose         show diff in commit message template\n",
      "\n",
      "Commit message options\n",
      "    -F, --file <file>     read message from file\n",
      "    --author <author>     override author for commit\n",
      "    --date <date>         override date for commit\n",
      "    -m, --message <message>\n",
      "                          commit message\n",
      "    -c, --reedit-message <commit>\n",
      "                          reuse and edit message from specified commit\n",
      "    -C, --reuse-message <commit>\n",
      "                          reuse message from specified commit\n",
      "    --fixup <commit>      use autosquash formatted message to fixup specified commit\n",
      "    --squash <commit>     use autosquash formatted message to squash specified commit\n",
      "    --reset-author        the commit is authored by me now (used with -C/-c/--amend)\n",
      "    -s, --signoff         add Signed-off-by:\n",
      "    -t, --template <file>\n",
      "                          use specified template file\n",
      "    -e, --edit            force edit of commit\n",
      "    --cleanup <default>   how to strip spaces and #comments from message\n",
      "    --status              include status in commit message template\n",
      "    -S, --gpg-sign[=<key-id>]\n",
      "                          GPG sign commit\n",
      "\n",
      "Commit contents options\n",
      "    -a, --all             commit all changed files\n",
      "    -i, --include         add specified files to index for commit\n",
      "    --interactive         interactively add files\n",
      "    -p, --patch           interactively add changes\n",
      "    -o, --only            commit only specified files\n",
      "    -n, --no-verify       bypass pre-commit and commit-msg hooks\n",
      "    --dry-run             show what would be committed\n",
      "    --short               show status concisely\n",
      "    --branch              show branch information\n",
      "    --ahead-behind        compute full ahead/behind values\n",
      "    --porcelain           machine-readable output\n",
      "    --long                show status in long format (default)\n",
      "    -z, --null            terminate entries with NUL\n",
      "    --amend               amend previous commit\n",
      "    --no-post-rewrite     bypass post-rewrite hook\n",
      "    -u, --untracked-files[=<mode>]\n",
      "                          show untracked files, optional modes: all, normal, no. (Default: all)\n",
      "\n",
      "error: src refspec master does not match any.\n",
      "error: failed to push some refs to 'https://github.com/Bbfxb2/-Class-1.7-and-1.8-.git'\n",
      "There is no tracking information for the current branch.\n",
      "Please specify which branch you want to merge with.\n",
      "See git-pull(1) for details.\n",
      "\n",
      "    git pull <remote> <branch>\n",
      "\n",
      "If you wish to set tracking information for this branch you can do so with:\n",
      "\n",
      "    git branch --set-upstream-to=<remote>/<branch> master\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "git init \n",
    "git add 1.8_find_git_stringreplacement_questions.ipynb\n",
    "git add 1.7_for_loop_string_replacement.ipynb\n",
    "git commit -\"first commit\"\n",
    "git push https://github.com/Bbfxb2/-Class-1.7-and-1.8-.git master\n",
    "git pull"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ".\n",
      "..\n",
      ".ipynb\n",
      ".ipynb_checkpoints\n",
      "1.3_jupyter-basics.ipynb\n",
      "1.4_working-with-files.ipynb\n",
      "1.5_redirection-and-pipes.ipynb\n",
      "1.6_awk.ipynb\n",
      "1.7_for_loop_string_replacement.ipynb\n",
      "1.8_find_git_stringreplacement_questions.ipynb\n",
      "Applied-Bioinformatics\n",
      "data\n",
      "file_list.txt\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls -a"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "find ./data/tmp -name \"*.txt\" -exec bash -c 'wc -l' _{} \\;"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "./file1.txt\n",
      "./file10.txt\n",
      "./file100.txt\n",
      "./file101.txt\n",
      "./file102.txt\n",
      "./file103.txt\n",
      "./file104.txt\n",
      "./file105.txt\n",
      "./file106.txt\n",
      "./file107.txt\n",
      "./file108.txt\n",
      "./file109.txt\n",
      "./file11.txt\n",
      "./file110.txt\n",
      "./file111.txt\n",
      "./file114.txt\n",
      "./file115.txt\n",
      "./file116.txt\n",
      "./file117.txt\n",
      "./file118.txt\n",
      "./file119.txt\n",
      "./file12.txt\n",
      "./file120.txt\n",
      "./file121.txt\n",
      "./file122.txt\n",
      "./file123.txt\n",
      "./file124.txt\n",
      "./file125.txt\n",
      "./file126.txt\n",
      "./file127.txt\n",
      "./file128.txt\n",
      "./file129.txt\n",
      "./file13.txt\n",
      "./file130.txt\n",
      "./file131.txt\n",
      "./file132.txt\n",
      "./file133.txt\n",
      "./file134.txt\n",
      "./file135.txt\n",
      "./file136.txt\n",
      "./file138.txt\n",
      "./file139.txt\n",
      "./file14.txt\n",
      "./file140.txt\n",
      "./file141.txt\n",
      "./file142.txt\n",
      "./file143.txt\n",
      "./file144.txt\n",
      "./file145.txt\n",
      "./file146.txt\n",
      "./file147.txt\n",
      "./file148.txt\n",
      "./file149.txt\n",
      "./file15.txt\n",
      "./file150.txt\n",
      "./file151.txt\n",
      "./file152.txt\n",
      "./file153.txt\n",
      "./file154.txt\n",
      "./file155.txt\n",
      "./file156.txt\n",
      "./file157.txt\n",
      "./file158.txt\n",
      "./file159.txt\n",
      "./file16.txt\n",
      "./file160.txt\n",
      "./file161.txt\n",
      "./file162.txt\n",
      "./file163.txt\n",
      "./file164.txt\n",
      "./file165.txt\n",
      "./file166.txt\n",
      "./file167.txt\n",
      "./file168.txt\n",
      "./file169.txt\n",
      "./file17.txt\n",
      "./file170.txt\n",
      "./file171.txt\n",
      "./file172.txt\n",
      "./file173.txt\n",
      "./file174.txt\n",
      "./file175.txt\n",
      "./file176.txt\n",
      "./file177.txt\n",
      "./file178.txt\n",
      "./file179.txt\n",
      "./file18.txt\n",
      "./file180.txt\n",
      "./file181.txt\n",
      "./file182.txt\n",
      "./file183.txt\n",
      "./file184.txt\n",
      "./file185.txt\n",
      "./file186.txt\n",
      "./file187.txt\n",
      "./file188.txt\n",
      "./file189.txt\n",
      "./file19.txt\n",
      "./file190.txt\n",
      "./file191.txt\n",
      "./file192.txt\n",
      "./file193.txt\n",
      "./file194.txt\n",
      "./file195.txt\n",
      "./file196.txt\n",
      "./file197.txt\n",
      "./file198.txt\n",
      "./file199.txt\n",
      "./file2.txt\n",
      "./file200.txt\n",
      "./file21.txt\n",
      "./file22.txt\n",
      "./file23.txt\n",
      "./file24.txt\n",
      "./file25.txt\n",
      "./file26.txt\n",
      "./file27.txt\n",
      "./file28.txt\n",
      "./file29.txt\n",
      "./file3.txt\n",
      "./file30.txt\n",
      "./file31.txt\n",
      "./file32.txt\n",
      "./file33.txt\n",
      "./file34.txt\n",
      "./file35.txt\n",
      "./file36.txt\n",
      "./file38.txt\n",
      "./file39.txt\n",
      "./file4.txt\n",
      "./file40.txt\n",
      "./file41.txt\n",
      "./file42.txt\n",
      "./file44.txt\n",
      "./file45.txt\n",
      "./file46.txt\n",
      "./file47.txt\n",
      "./file48.txt\n",
      "./file49.txt\n",
      "./file5.txt\n",
      "./file50.txt\n",
      "./file51.txt\n",
      "./file52.txt\n",
      "./file53.txt\n",
      "./file54.txt\n",
      "./file55.txt\n",
      "./file56.txt\n",
      "./file57.txt\n",
      "./file58.txt\n",
      "./file59.txt\n",
      "./file6.txt\n",
      "./file60.txt\n",
      "./file61.txt\n",
      "./file62.txt\n",
      "./file64.txt\n",
      "./file65.txt\n",
      "./file66.txt\n",
      "./file67.txt\n",
      "./file68.txt\n",
      "./file69.txt\n",
      "./file7.txt\n",
      "./file70.txt\n",
      "./file71.txt\n",
      "./file72.txt\n",
      "./file73.txt\n",
      "./file74.txt\n",
      "./file75.txt\n",
      "./file76.txt\n",
      "./file77.txt\n",
      "./file78.txt\n",
      "./file79.txt\n",
      "./file8.txt\n",
      "./file80.txt\n",
      "./file81.txt\n",
      "./file82.txt\n",
      "./file84.txt\n",
      "./file85.txt\n",
      "./file86.txt\n",
      "./file87.txt\n",
      "./file88.txt\n",
      "./file89.txt\n",
      "./file9.txt\n",
      "./file90.txt\n",
      "./file91.txt\n",
      "./file92.txt\n",
      "./file93.txt\n",
      "./file94.txt\n",
      "./file96.txt\n",
      "./file97.txt\n",
      "./file98.txt\n",
      "./file99.txt\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd ./data/tmp/\n",
    "find ./ -name \"*.txt\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0 ./data/tmp/file1.txt\n",
      "0 ./data/tmp/file10.txt\n",
      "0 ./data/tmp/file100.txt\n",
      "0 ./data/tmp/file101.txt\n",
      "0 ./data/tmp/file102.txt\n",
      "0 ./data/tmp/file103.txt\n",
      "0 ./data/tmp/file104.txt\n",
      "0 ./data/tmp/file105.txt\n",
      "0 ./data/tmp/file106.txt\n",
      "0 ./data/tmp/file107.txt\n",
      "0 ./data/tmp/file108.txt\n",
      "0 ./data/tmp/file109.txt\n",
      "0 ./data/tmp/file11.txt\n",
      "0 ./data/tmp/file110.txt\n",
      "0 ./data/tmp/file111.txt\n",
      "0 ./data/tmp/file114.txt\n",
      "0 ./data/tmp/file115.txt\n",
      "0 ./data/tmp/file116.txt\n",
      "0 ./data/tmp/file117.txt\n",
      "0 ./data/tmp/file118.txt\n",
      "0 ./data/tmp/file119.txt\n",
      "0 ./data/tmp/file12.txt\n",
      "0 ./data/tmp/file120.txt\n",
      "0 ./data/tmp/file121.txt\n",
      "0 ./data/tmp/file122.txt\n",
      "0 ./data/tmp/file123.txt\n",
      "0 ./data/tmp/file124.txt\n",
      "0 ./data/tmp/file125.txt\n",
      "0 ./data/tmp/file126.txt\n",
      "0 ./data/tmp/file127.txt\n",
      "0 ./data/tmp/file128.txt\n",
      "0 ./data/tmp/file129.txt\n",
      "0 ./data/tmp/file13.txt\n",
      "0 ./data/tmp/file130.txt\n",
      "0 ./data/tmp/file131.txt\n",
      "0 ./data/tmp/file132.txt\n",
      "0 ./data/tmp/file133.txt\n",
      "0 ./data/tmp/file134.txt\n",
      "0 ./data/tmp/file135.txt\n",
      "0 ./data/tmp/file136.txt\n",
      "0 ./data/tmp/file138.txt\n",
      "0 ./data/tmp/file139.txt\n",
      "0 ./data/tmp/file14.txt\n",
      "0 ./data/tmp/file140.txt\n",
      "0 ./data/tmp/file141.txt\n",
      "0 ./data/tmp/file142.txt\n",
      "0 ./data/tmp/file143.txt\n",
      "0 ./data/tmp/file144.txt\n",
      "0 ./data/tmp/file145.txt\n",
      "0 ./data/tmp/file146.txt\n",
      "0 ./data/tmp/file147.txt\n",
      "0 ./data/tmp/file148.txt\n",
      "0 ./data/tmp/file149.txt\n",
      "0 ./data/tmp/file15.txt\n",
      "0 ./data/tmp/file150.txt\n",
      "0 ./data/tmp/file151.txt\n",
      "0 ./data/tmp/file152.txt\n",
      "0 ./data/tmp/file153.txt\n",
      "0 ./data/tmp/file154.txt\n",
      "0 ./data/tmp/file155.txt\n",
      "0 ./data/tmp/file156.txt\n",
      "0 ./data/tmp/file157.txt\n",
      "0 ./data/tmp/file158.txt\n",
      "0 ./data/tmp/file159.txt\n",
      "0 ./data/tmp/file16.txt\n",
      "0 ./data/tmp/file160.txt\n",
      "0 ./data/tmp/file161.txt\n",
      "0 ./data/tmp/file162.txt\n",
      "0 ./data/tmp/file163.txt\n",
      "0 ./data/tmp/file164.txt\n",
      "0 ./data/tmp/file165.txt\n",
      "0 ./data/tmp/file166.txt\n",
      "0 ./data/tmp/file167.txt\n",
      "0 ./data/tmp/file168.txt\n",
      "0 ./data/tmp/file169.txt\n",
      "0 ./data/tmp/file17.txt\n",
      "0 ./data/tmp/file170.txt\n",
      "0 ./data/tmp/file171.txt\n",
      "0 ./data/tmp/file172.txt\n",
      "0 ./data/tmp/file173.txt\n",
      "0 ./data/tmp/file174.txt\n",
      "0 ./data/tmp/file175.txt\n",
      "0 ./data/tmp/file176.txt\n",
      "0 ./data/tmp/file177.txt\n",
      "0 ./data/tmp/file178.txt\n",
      "0 ./data/tmp/file179.txt\n",
      "0 ./data/tmp/file18.txt\n",
      "0 ./data/tmp/file180.txt\n",
      "0 ./data/tmp/file181.txt\n",
      "0 ./data/tmp/file182.txt\n",
      "0 ./data/tmp/file183.txt\n",
      "0 ./data/tmp/file184.txt\n",
      "0 ./data/tmp/file185.txt\n",
      "0 ./data/tmp/file186.txt\n",
      "0 ./data/tmp/file187.txt\n",
      "0 ./data/tmp/file188.txt\n",
      "0 ./data/tmp/file189.txt\n",
      "0 ./data/tmp/file19.txt\n",
      "0 ./data/tmp/file190.txt\n",
      "0 ./data/tmp/file191.txt\n",
      "0 ./data/tmp/file192.txt\n",
      "0 ./data/tmp/file193.txt\n",
      "0 ./data/tmp/file194.txt\n",
      "0 ./data/tmp/file195.txt\n",
      "0 ./data/tmp/file196.txt\n",
      "0 ./data/tmp/file197.txt\n",
      "0 ./data/tmp/file198.txt\n",
      "0 ./data/tmp/file199.txt\n",
      "0 ./data/tmp/file2.txt\n",
      "0 ./data/tmp/file200.txt\n",
      "0 ./data/tmp/file21.txt\n",
      "0 ./data/tmp/file22.txt\n",
      "0 ./data/tmp/file23.txt\n",
      "0 ./data/tmp/file24.txt\n",
      "0 ./data/tmp/file25.txt\n",
      "0 ./data/tmp/file26.txt\n",
      "0 ./data/tmp/file27.txt\n",
      "0 ./data/tmp/file28.txt\n",
      "0 ./data/tmp/file29.txt\n",
      "0 ./data/tmp/file3.txt\n",
      "0 ./data/tmp/file30.txt\n",
      "0 ./data/tmp/file31.txt\n",
      "0 ./data/tmp/file32.txt\n",
      "0 ./data/tmp/file33.txt\n",
      "0 ./data/tmp/file34.txt\n",
      "0 ./data/tmp/file35.txt\n",
      "0 ./data/tmp/file36.txt\n",
      "0 ./data/tmp/file38.txt\n",
      "0 ./data/tmp/file39.txt\n",
      "0 ./data/tmp/file4.txt\n",
      "0 ./data/tmp/file40.txt\n",
      "0 ./data/tmp/file41.txt\n",
      "0 ./data/tmp/file42.txt\n",
      "0 ./data/tmp/file44.txt\n",
      "0 ./data/tmp/file45.txt\n",
      "0 ./data/tmp/file46.txt\n",
      "0 ./data/tmp/file47.txt\n",
      "0 ./data/tmp/file48.txt\n",
      "0 ./data/tmp/file49.txt\n",
      "0 ./data/tmp/file5.txt\n",
      "0 ./data/tmp/file50.txt\n",
      "0 ./data/tmp/file51.txt\n",
      "0 ./data/tmp/file52.txt\n",
      "0 ./data/tmp/file53.txt\n",
      "0 ./data/tmp/file54.txt\n",
      "0 ./data/tmp/file55.txt\n",
      "0 ./data/tmp/file56.txt\n",
      "0 ./data/tmp/file57.txt\n",
      "0 ./data/tmp/file58.txt\n",
      "0 ./data/tmp/file59.txt\n",
      "0 ./data/tmp/file6.txt\n",
      "0 ./data/tmp/file60.txt\n",
      "0 ./data/tmp/file61.txt\n",
      "0 ./data/tmp/file62.txt\n",
      "0 ./data/tmp/file64.txt\n",
      "0 ./data/tmp/file65.txt\n",
      "0 ./data/tmp/file66.txt\n",
      "0 ./data/tmp/file67.txt\n",
      "0 ./data/tmp/file68.txt\n",
      "0 ./data/tmp/file69.txt\n",
      "0 ./data/tmp/file7.txt\n",
      "0 ./data/tmp/file70.txt\n",
      "0 ./data/tmp/file71.txt\n",
      "0 ./data/tmp/file72.txt\n",
      "0 ./data/tmp/file73.txt\n",
      "0 ./data/tmp/file74.txt\n",
      "0 ./data/tmp/file75.txt\n",
      "0 ./data/tmp/file76.txt\n",
      "0 ./data/tmp/file77.txt\n",
      "0 ./data/tmp/file78.txt\n",
      "0 ./data/tmp/file79.txt\n",
      "0 ./data/tmp/file8.txt\n",
      "0 ./data/tmp/file80.txt\n",
      "0 ./data/tmp/file81.txt\n",
      "0 ./data/tmp/file82.txt\n",
      "0 ./data/tmp/file84.txt\n",
      "0 ./data/tmp/file85.txt\n",
      "0 ./data/tmp/file86.txt\n",
      "0 ./data/tmp/file87.txt\n",
      "0 ./data/tmp/file88.txt\n",
      "0 ./data/tmp/file89.txt\n",
      "0 ./data/tmp/file9.txt\n",
      "0 ./data/tmp/file90.txt\n",
      "0 ./data/tmp/file91.txt\n",
      "0 ./data/tmp/file92.txt\n",
      "0 ./data/tmp/file93.txt\n",
      "0 ./data/tmp/file94.txt\n",
      "0 ./data/tmp/file96.txt\n",
      "0 ./data/tmp/file97.txt\n",
      "0 ./data/tmp/file98.txt\n",
      "0 ./data/tmp/file99.txt\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "find ./data/tmp -name \"*.txt\" -exec bash -c 'wc -l $1' _ {} \\;"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "HW\n",
    "Count number of characters in the filename of each .txt file in ./data/tmp/ folder.\n",
    "Print all nucleotide sequences from ./data/fastq/simulated1.R1.fq by replacing all 'T's in the nucleotide sequences with 'U's. Hint: ${variable//123/112} will replace all occurences of 123 with 112."
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
